packages/discord.js/src/util/Components.js

```diff
@@ -89,19 +156,64 @@ function createComponent(data) {
       return new MentionableSelectMenuComponent(data);
     case ComponentType.ChannelSelect:
       return new ChannelSelectMenuComponent(data);
+    case ComponentType.Container:
+      return new ContainerComponent(data);
+    case ComponentType.TextDisplay:
+      return new TextDisplayComponent(data);
+    case ComponentType.File:
+      return new FileComponent(data);
+    case ComponentType.MediaGallery:
+      return new MediaGalleryComponent(data);
+    case ComponentType.Section:
+      return new SectionComponent(data);
+    case ComponentType.Separator:
+      return new SeparatorComponent(data);
+    case ComponentType.Thumbnail:
+      return new ThumbnailComponent(data);
     default:
       return new Component(data);
   }
 }
 
+/**
+ * Finds a component by customId in nested components
+ * @param {Array<Component|APIMessageComponent>} components The components to search in
+ * @param {string} customId The customId to search for
+ * @returns {Component|APIMessageComponent}
+ */
+function findComponentByCustomId(components, customId) {
+  return (
+    components
+      .flatMap(component => {
+        switch (component.type) {
+          case ComponentType.ActionRow:
+            return component.components;
+          case ComponentType.Section:
+            return [component.accessory];
```

Comment:
Shouldn't this be `[...component.components, component.accessory]`

Reply:
😅🥲 yes

Reply:
```suggestion
            return [...component.components, component.accessory];
```